### PR TITLE
database/pg: remove Begin

### PR DESCRIPTION
**Pull request by jbowens**
Remove pg.Begin. Call sites must explicitly check that a DB handle
implements Beginner.

### CODE REVIEW

**Comment by jbowens** [comment]
PTAL. I'm not actually sure what the original use of this code was, but I assume it was for using pgtest.NewTx in tests that tested code already using SQL transactions?

File: database/pg/context.go [hunk 1]
```diff
@@ -53,13 +53,12 @@ var dbKey key
 // associated database.
 //
 // Note: if a transaction is already pending in the passed-in context,
-// this function does not create a new one but returns the existing
-// one.
+// this function panics.
 func Begin(ctx context.Context) (Committer, context.Context, error) {
```
Comment (kr) [inline database/pg/context.go]: Are we at the point yet where we can delete this function entirely? It is one of the "context convenience functions".

**Comment by kr** [comment]
LGTM

File: database/pg/context.go (hunk 1 above)
Comment (jbowens) [inline database/pg/context.go]: > Are we at the point yet where we can delete this function entirely? It is one of the "context convenience functions".

Good call. There are there call sites, all in `core/account/utxodb`. I'll remove it in this PR.

**Comment by kr** [comment]
> I assume it was for using pgtest.NewTx in tests that tested code already using SQL transactions?

Yeah, exactly.

File: core/account/utxodb/reserve.go [hunk 2]
```diff
@@ -57,15 +58,23 @@ type (
 	}
 )
 
+func beginTxn(ctx context.Context) (pg.Tx, error) {
+	db := pg.FromContext(ctx)
+	beginner, ok := db.(pg.Beginner)
```
Comment (kr) [inline core/account/utxodb/reserve.go]: I think pg.Beginner isn't the interface we need here. A real sql.DB has a different signature. See the deleted pg.begin:

```
type beginner interface {		
	Begin(context.Context) (*sql.Tx, error)		
}		
switch d := db.(type) {		
case beginner: // e.g. *sql.DB		
	return d.Begin(ctx)		
case Beginner: // e.g. pgtest.noCommitDB		
	return d.Begin(ctx)		
}
```

Comment (kr) [inline core/account/utxodb/reserve.go]: I think maybe we should type assert to the concrete type `*sql.DB` inline at the call sites.

**Pull request merged**